src/composables/transferSendForm.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  computed,
+  ref,
+} from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useForm } from 'vee-validate';
+
+import {
+  Dictionary,
+  IAsset,
+  IToken,
+  TransferFormModel,
+} from '@/types';
+import { getMessageByFieldName } from '@/popup/utils/transferSend';
+import { useModals } from '@/composables/modals';
+import {
+  APP_LINK_WEB,
+  MODAL_READ_QR_CODE,
+} from '@/constants';
+import { toShiftedBigNumber } from '@/utils';
+
+type SelectedAssetValueFunction = (token?: string, selectedAsset?: IAsset) => IToken | undefined;
+
+interface UseTransferSendFormParams {
+  transferData: TransferFormModel;
+  getSelectedAssetValue?: SelectedAssetValueFunction;
+}
+
+export function useTransferSendForm({
+  transferData,
+  getSelectedAssetValue,
+}: UseTransferSendFormParams) {
+  const formModel = ref(transferData);
+  const invoiceId = ref(null);
+  const invoiceContract = ref(null);
+
+  const { t } = useI18n();
+  const { openModal, openDefaultModal } = useModals();
+  const { errors, validate } = useForm();
+
+  const hasError = computed((): boolean => ['address', 'amount'].some((errorKey) => getMessageByFieldName(errors.value[errorKey]) === 'error'));
+
+  function clearPayload() {
+    formModel.value.payload = '';
+  }
+
+  function handleAssetChange(selectedAsset: IToken) {
+    formModel.value.selectedAsset = selectedAsset;
+  }
+
+  function returnFormModelValues({
+    account,
+    amount,
+    payload,
+    token,
+  }: Dictionary, tokenHandler?: SelectedAssetValueFunction) {
+    const result: Partial<TransferFormModel> = {};
+
+    const selectedToken = tokenHandler ? tokenHandler(token, formModel.value.selectedAsset) : null;
+    if (selectedToken) {
+      result.selectedAsset = selectedToken;
+    }
+
+    if (account) {
+      result.address = account;
+    }
+    if (amount) {
+      result.amount = amount;
+    }
+    if (payload) {
+      result.payload = payload;
+    }
+
+    return result;
+  }
+
+  function updateFormModelValues(params: Dictionary) {
+    formModel.value = {
+      ...formModel.value,
+      ...returnFormModelValues(params, getSelectedAssetValue),
+    };
+  }
+
+  async function openScanQrModal(tokenBalances: IToken[]) {
+    const scanResult = await openModal(MODAL_READ_QR_CODE, {
+      title: t('pages.send.scanAddress'),
+      icon: 'critical',
+    });
+    if (scanResult?.trim().charAt(0) === '{') {
+      let parsedScanResult: any = null;
+      try {
+        parsedScanResult = JSON.parse(scanResult);
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        if (process.env.NODE_ENV !== 'production') {
```

Comment:
Please create the `IS_PRODUCTION` constant in the `environment.ts` file and import it here.